internal/kibana/agents.go

```diff
@@ -99,13 +102,19 @@ func (c *Client) waitUntilPolicyAssigned(a Agent, p Policy) error {
 		}
 		logger.Debugf("Agent data: %s", agent.String())
 
-		if agent.PolicyID == p.ID && agent.PolicyRevision == p.Revision {
+		if agent.PolicyID == p.ID && agent.PolicyRevision >= p.Revision {
```

Comment:
In my latest runs of elastic-package test in different packages locally, I didn't see the message (error?) of this timeout.

Is the Revision number always increased ?

Reply:
I have seen this when trying https://github.com/elastic/elastic-package/pull/1274 locally. The Agent has the policy assigned, but it has a newer revision, then it is never satisfied and the cleanup is interrupted with a timeout error after a long wait. Not sure though how I reached this situation, but it happened to me at least a couple of times.

In this situation the only options are to accept the newer policy or to fail fast, and I think that accepting the policy as is now is the best option of these two. There does not seem to be any way to assign an specific revision of a policy.

Reply:
> There does not seem to be any way to assign an specific revision of a policy.

There isn't, checked in https://petstore.swagger.io/?url=https://raw.githubusercontent.com/elastic/kibana/8.7/x-pack/plugins/fleet/common/openapi/bundled.json#/default/reassign-agent

Reply:
> Is the Revision number always increased ?

I am hoping so :smiley:

Reply:
Perfect! Good catch!